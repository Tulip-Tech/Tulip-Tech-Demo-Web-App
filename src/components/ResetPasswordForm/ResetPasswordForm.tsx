import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './ResetPasswordForm.module.scss';

import Button from '#components/Button/Button';

type Props = {
  onCancel: () => void;
  onReset: () => void;
  submitting: boolean;
};

const ResetPasswordForm: React.FC<Props> = ({ onCancel, onReset, submitting }: Props) => {
  const { t } = useTranslation('account');
  return (
    <div className={styles.resetPassword}>
      <h5 className={styles.title}>{t('reset.reset_password')}</h5>
      <p className={styles.text}>{t('reset.text')}</p>
      <Button onClick={onReset} className={styles.button} fullWidth color="primary" label={t('reset.yes')} disabled={submitting} />
      <Button onClick={onCancel} fullWidth label={t('reset.no')} />
    </div>
  );
};

export default ResetPasswordForm;
