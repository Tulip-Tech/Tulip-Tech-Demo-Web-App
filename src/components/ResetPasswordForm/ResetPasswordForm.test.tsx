import React from 'react';
import { render } from '@testing-library/react';

import ResetPasswordForm from './ResetPasswordForm';

describe('<ResetPassword>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ResetPasswordForm onCancel={vi.fn()} onReset={vi.fn()} submitting={false} />);

    expect(container).toMatchSnapshot();
  });
});
