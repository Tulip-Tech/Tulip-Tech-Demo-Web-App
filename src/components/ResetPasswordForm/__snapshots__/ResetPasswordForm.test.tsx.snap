// Vitest Snapshot v1

exports[`<ResetPassword> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h5
      class="_title_1976d9"
    >
      reset.reset_password
    </h5>
    <p
      class="_text_1976d9"
    >
      reset.text
    </p>
    <button
      aria-disabled="false"
      class="_button_f8f296 _button_1976d9 _primary_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        reset.yes
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        reset.no
      </span>
    </button>
  </div>
</div>
`;
